vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.ts

```diff
@@ -364,19 +362,15 @@ window.Vaadin.Flow.gridConnector.initLazy = (grid) => {
     currentUpdateSetRange = [firstPage, firstPage + updatedPageCount - 1];
 
     for (let i = 0; i < updatedPageCount; i++) {
-      let page = firstPage + i;
-      let slice = items.slice(i * grid.pageSize, (i + 1) * grid.pageSize);
-      cache[page] = slice;
-
-      grid.$connector.doSelection(slice.filter((item) => item.selected));
-      grid.$connector.doDeselection(slice.filter((item) => !item.selected && selectedKeys[item.key]));
-
-      const updatedItems = updateGridCache(page);
-      if (updatedItems) {
-        itemsUpdated(updatedItems);
-      }
+      const page = firstPage + i;
+      cache[page] = items.slice(i * grid.pageSize, (i + 1) * grid.pageSize);
+      updateGridCache(page);
     }
 
+    itemsUpdated(items);
+    grid.$connector.doSelection(items.filter((item) => item.selected));
```

Comment:
Is there a reason for changing the order of updating items and updating selection?